crates/block-explorers/src/lib.rs

```diff
@@ -241,17 +241,20 @@ impl Client {
         })?;
 
         match res {
-            ResponseData::Error { result, message, status } => {
+            ResponseData::Error { message, result } => {
                 if let Some(ref result) = result {
                     if result.starts_with("Max rate limit reached") {
                         return Err(EtherscanError::RateLimitExceeded);
-                    } else if result.to_lowercase() == "invalid api key" {
+                    }
+                    if result.to_lowercase().contains("invalid api key") {
```

Comment:
could you please submit this change separately?

Reply:
Hey @mattsse, I'd be happy to but the reality is that it is a dead case without the change to the deserializer. The test case and the original case where `result == "invalid api key"` can never be hit since it would deserialize as a `ResponseData::Success` case rathere than `ResponseData::Error`

Let me know if you would still like it as a separate change?